packages/wallets/wallet-evm/src/strategy/strategy.ts

```diff
@@ -64,6 +69,41 @@ export class EvmWallet
       )
     }
 
+    if (!isServerSide()) {
+      window.addEventListener(
+        'eip6963:announceProvider',
+        (announcement: any) => {
+          const event = announcement as unknown as EIP6963AnnounceProviderEvent
+          const walletName = event.detail.info.name.toLowerCase()
+
+          if (walletName === Wallet.Metamask.toLowerCase()) {
+            this.evmProviders[Wallet.Metamask] = event.detail.provider
+          }
+
+          if (walletName === Wallet.Rainbow.toLowerCase()) {
+            this.evmProviders[Wallet.Rainbow] = event.detail.provider
+          }
+
+          if (walletName === Wallet.Phantom.toLowerCase()) {
+            this.evmProviders[Wallet.Phantom] = event.detail.provider
+          }
+
+          if (walletName === Wallet.OkxWallet.toLowerCase()) {
+            this.evmProviders[Wallet.OkxWallet] = event.detail.provider
+          }
+
+          if (walletName === Wallet.BitGet.toLowerCase()) {
+            this.evmProviders[Wallet.BitGet] = event.detail.provider
+          }
+
+          if (walletName === Wallet.TrustWallet.toLowerCase()) {
+            this.evmProviders[Wallet.TrustWallet] = event.detail.provider
+          }
+        },
+      )
+      window.dispatchEvent(new Event('eip6963:requestProvider'))
+    }
+
```

Comment:
@bangjelkoski, this is the fix for the wallet race condition.
i.e solana and rainbow overriding metamask.

Can you take a look and let us know what you think about this?